### PR TITLE
[SHIPA-1022] Registry Auth fails on Mac

### PR DESCRIPTION
**Pull request by murphybytes**
# Description

See [SHIPA-1022](https://shipaio.atlassian.net/browse/SHIPA-1022?atlOrigin=eyJpIjoiODhiM2UwY2ZkZDU2NGIyZmJjZTVlMjI4NDE3MDlhZjgiLCJwIjoiaiJ9). Ketch uses the same mechanism to obtain registry authentication credentials as the Docker CLI.  It relies on the Docker CLI being present in order to deploy applications from source. Docker needs to shell out to an operating system specific helper function that retrieves auth credentials from secret storage for the OS, for example, the keychain on the Mac. We were not handling that correctly.

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Chore (documentation addition or typo, file relocation)

## Testing

- [x] New tests were added with this PR that prove my fix is effective or that my feature works (describe below this bullet)
- [ ] This change requires no testing (i.e. documentation update)

## Documentation

- [x] All added public packages, funcs, and types have been documented with doc comments
- [x] I have commented my code, particularly in hard-to-understand areas

## Final Checklist:

- [x] I followed standard [GitHub flow](https://guides.github.com/introduction/flow/) guidelines
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

## Additional Information (omit if empty)

Please include anything else relevant to this PR that may be useful to know.

### CODE REVIEW

File: internal/docker/auth.go [hunk 1]
```diff
@@ -2,50 +2,42 @@ package docker
...
-			Auth:          v.Auth,
-			Email:         v.Email,
-			ServerAddress: v.ServerAddress,
-			IdentityToken: v.IdentityToken,
-			RegistryToken: v.RegistryToken,
-		}
-
-	}
-	return result
+func official(regHost string) bool {
```
Comment (parkerduckworth) [inline internal/docker/auth.go:41]: I don't personally have a problem with this, but some people prefer to see an `is` suffix for `bool` returning funcs.
Ex: `isHostOfficial(regHost)`

File: internal/docker/docker.go [hunk 2]
```diff
@@ -6,21 +6,17 @@ import (
...
 	"encoding/base64"
-	"io"
-	"os/user"
-	"path"
-
 	"github.com/docker/distribution/reference"
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/client"
 	"github.com/docker/docker/pkg/archive"
+	"io"
```
Comment (parkerduckworth) [inline internal/docker/docker.go:13]: stdlib import should be with its counterparts

File: internal/docker/auth.go [hunk 3]
```diff
@@ -2,50 +2,42 @@ package docker
```
Comment (shuminyang) [inline internal/docker/auth.go:2]: Noticed this package doesn't have documentation, since you're working on this might as well put docs on it

Comment (murphybytes) [reply internal/docker/auth.go:2]: @shuminyang it does have package docs in the docker.go file

**Review by parkerduckworth — APPROVED** [review]

**Pull request merged**